### PR TITLE
Save revel key/value when building variant objects

### LINKED ISSUE #3935: REVEL score not added to variant_obj in build_variant from parsed variants.
Hi,

I'm new to scout and ran into an issue with scout only displaying REVEL rank scores, but not REVEL scores when trying to add REVEL annotations to one of our pipelines. 

If I read the code correctly, `parse_variant` is where both the revel score and rank score are pulled out of the vep annotations into a parsed variant object:

https://github.com/Clinical-Genomics/scout/blob/2373185c2c9d71cd1bbdaaba9add232d316612d1/scout/parse/variant/variant.py#L229-L238

However, it seems that only the rank score is used when the parsed variant object is passed to the function that builds the database variant object:

https://github.com/Clinical-Genomics/scout/blob/2373185c2c9d71cd1bbdaaba9add232d316612d1/scout/build/variant/variant.py#L243-L246

i.e. there's a ` variant_obj["revel"] = variant.get("revel")` missing in the code block above. Is this correct?

I've of course confirmed that both REVEL scores and rank scores are correctly added to the VCFs, with correct field names etc.

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #3935 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
I'm going to load a case on stage to see if it works!

**Comment by northwestwitch** [comment]
Mm I wonder if what we display is correct. According to the code, what we collect and save under `revel` is the [revel rank score](https://github.com/Clinical-Genomics/scout/blob/2373185c2c9d71cd1bbdaaba9add232d316612d1/scout/parse/variant/transcript.py#L43)

But we switch these values in the variant page? Look at the one that was missing and re-appears with this PR!

**Comment by northwestwitch** [comment]
After switching the 2 values in the variant view:

image

**Comment by dnil** [comment]
> Mm I wonder if what we display is correct. According to the code, what we collect and save under `revel` is the [revel rank score](https://github.com/Clinical-Genomics/scout/blob/2373185c2c9d71cd1bbdaaba9add232d316612d1/scout/parse/variant/transcript.py#L43)
> 
> But we switch these values in the variant page? Look at the one that was missing and re-appears with this PR!

I think you may have tried to cater for legacy cases (pre the MIP patch)? @jemten just noted the switch here https://github.com/Clinical-Genomics/MIP/issues/1981.

**Comment by northwestwitch** [comment]
> We're still annotating the vcf with both values but only REVEL_score is used for ranking with genmod

Good, since Scout shows both. But I still think that their values are switched in our variant view

File: scout/server/blueprints/variant/templates/variant/variant_details.html [hunk 1]
```diff
@@ -250,8 +250,8 @@
           <a href="http://database.liulab.science/dbNSFP" target="_blank" data-bs-toggle="tooltip"
              title="REVEL scores were ranked among all REVEL scores in dbNSFP. The REVEL rank score is the ratio of the rank of the score over the total number of REVEL scores in dbNSFP">REVEL rank score</a>
           <span class="float-end">
-          {% if variant.revel_score  %}
-             {{ variant.revel_score }}
+          {% if variant.revel  %}
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/variant_details.html:253]: Hm, but this does not make sense? We used the revel rank score exclusively before, named `revel_score`. The raw score was added after Hans remarked some guidelines used it.

Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/variant_details.html:253]: Ah, not `revel_score` everywhere.. It swaps names.. 😆

File: scout/build/variant/variant.py [hunk 2]
```diff
@@ -105,6 +105,7 @@ def build_variant(
         # Predicted deleteriousness:
         cadd_score = float,
         revel_score = float,
+        revel = float
```
Comment (dnil) [inline scout/build/variant/variant.py:108]: 👍 And perhaps a comment is in order to indicate this is the raw score..

File: scout/build/variant/variant.py [hunk 3]
```diff
@@ -243,6 +244,7 @@ def build_variant(
     ##### Add the severity predictors #####
     variant_obj["cadd_score"] = variant.get("cadd_score")
     variant_obj["revel_score"] = variant.get("revel_score")
+    variant_obj["revel"] = variant.get("revel")
```
Comment (dnil) [inline scout/build/variant/variant.py:247]: 💯

File: scout/server/blueprints/variant/templates/variant/variant_details.html (hunk 1 above)
Comment (northwestwitch) [reply scout/server/blueprints/variant/templates/variant/variant_details.html:253]: I'm confused. Where the variant is parsed the code is like this:

image

image

They seem swapped in tto me in the template? 🤔

Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/variant_details.html:253]: Ah, but no. The transcript has `REVEL_RANKSCORE` stored as `revel` (parse/variant/transcript.py:43). Then the variant has it as `revel_score` parse/variant/transcript.py:239. 

Next time, more informative variable names, and no swaps with an old name. 😬

**Review by dnil — CHANGES_REQUESTED** [review]

File: scout/server/blueprints/variant/templates/variant/variant_details.html (hunk 1 above)
Comment (northwestwitch) [reply scout/server/blueprints/variant/templates/variant/variant_details.html:253]: Ah right I missed when we swapped them in the parse/variant/transcriipt 🤣 

image

But perhaps to make it easier by swapping the keys where they are first created (parse/variant/transcript.py:43). Let's see

Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/variant_details.html:253]: Yes, it is not exactly kind to the reader. 😆 Just don't go changing the keys for anything that gets stored on db; we definitely don't also want to have legacy cases around with another nomenclature.

Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/variant_details.html:253]: Or, well, if so, make it a completely new name so we can ask for both.

**Comment by dnil** [comment]
...and to reconfirm, you will find that the REVEL_RANKSCORE for https://scout-stage.scilifelab.se/cust002/F0046714-2/2a536f42479c4d861df98599beee8e62 in the VCF has `|0.98973|0.947|` with CSQ key `|REVEL_rankscore|REVEL_score|`. So currently swapped on stage, but as soon as you deploy this latest revert should look all good. 😸

**Comment by dnil** [comment]
For the record, deploying again with the reverted branch looks all consistent with the VCF, so no reason for worry about old cases:
Screenshot 2023-05-12 at 08 01 24

That does not say you can't refactor if you want. 😄

File: scout/parse/variant/transcript.py [hunk 4]
```diff
@@ -41,7 +41,7 @@ def parse_transcripts(raw_transcripts):
         transcript["sift_prediction"] = get_sift_prediction(entry)
 
         if entry.get("REVEL_RANKSCORE"):
-            transcript["revel"] = float(entry.get("REVEL_RANKSCORE"))
+            transcript["revel_rankscore"] = float(entry.get("REVEL_RANKSCORE"))
```
Comment (dnil) [inline scout/parse/variant/transcript.py:44]: 💯

**Review by dnil — APPROVED** [review]
Lets await the test as well, but this looks fine!

**Comment by northwestwitch** [comment]
Tested the latest changes on Hasta by re-uploading the same case. All good. Merging!

**Pull request merged**